bundles/org.openhab.binding.airq/src/main/java/org/openhab/binding/airq/internal/AirqHandler.java

```diff
@@ -0,0 +1,797 @@
+        byte[] IV = new byte[16];
+        SecureRandom random = new SecureRandom();
+        random.nextBytes(IV);
+        SecretKey seckey = new SecretKeySpec(passkey, 0, passkey.length, "AES");
+        try {
+            Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
+            SecretKeySpec keySpec = new SecretKeySpec(seckey.getEncoded(), "AES");
+            IvParameterSpec ivSpec = new IvParameterSpec(IV);
+            cipher.init(Cipher.ENCRYPT_MODE, keySpec, ivSpec);
+            byte[] encryptedText = cipher.doFinal(toencode);
+            byte[] totaltext = new byte[16 + encryptedText.length];
+            System.arraycopy(IV, 0, totaltext, 0, 16);
+            System.arraycopy(encryptedText, 0, totaltext, 16, encryptedText.length);
+            byte[] encodedcontent = Base64.getEncoder().encode(totaltext);
+            logger.trace("air-Q - airqHandler - encrypt(): encrypted text: {}", encodedcontent);
+            content = new String(encodedcontent);
+        } catch (Exception e) {
+            logger.warn("air-Q - airqHandler - encrypt(): Error while encrypting: {}", e.toString());
+        }
+        return content;
+    }
+
+    // gets the data after online/offline management and does the JSON work, or at least the first step.
+    protected @Nullable String getDecryptedContentString(String url, String requestMethod, @Nullable String body) {
+        Result res = null;
+        String jsonAnswer = null;
+        res = getData(url, "GET", null);
+        if (res != null) {
+            String jsontext = res.getBody();
+            logger.trace("air-Q - airqHandler - getDecryptedContentString(): Result from getData() is {} with body={}",
+                    res, res.getBody());
+            // Gson code based on https://riptutorial.com/de/gson
+            Gson gson = new Gson();
+            JsonElement ans = gson.fromJson(jsontext, JsonElement.class);
+            if (ans != null) {
+                JsonObject jsonObj = ans.getAsJsonObject();
+                jsonAnswer = decrypt(jsonObj.get("content").getAsString().getBytes(), config.password);
+                if (jsonAnswer == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Decryption not possible, probably wrong password");
+                }
+            } else {
+                logger.warn(
+                        "air-Q - airqHandler - getDecryptedContentString(): The air-Q data could not be extracted from this string: {}",
+                        ans);
+            }
+        }
+        return jsonAnswer;
+    }
+
+    // calls the networking job and in addition does additional tests for online/offline management
+    protected @Nullable Result getData(String address, String requestMethod, @Nullable String body) {
+        Result res = null;
+        res = doNetwork(address, "GET", body);
+        if (res == null) {
+            if (thStatus != ThingStatus.OFFLINE) {
```

Comment:
Is there any reason for caching the status? You can retrieve it with `getThing().getStatus()`.